src/NLog/Logger.cs

```diff
@@ -95,14 +96,35 @@ public bool IsEnabled(LogLevel level)
             return this.GetTargetsForLevel(level) != null;
         }
 
+#if NET4_5 
         /// <summary>
         /// Writes the specified diagnostic message.
         /// </summary>
         /// <param name="logEvent">Log event.</param>
-        public void Log(LogEventInfo logEvent)
+        /// <param name="memberName">Method or property name of the caller.</param>
+        /// <param name="sourceFilePath">Full path of the source file that contains the caller.</param>
+        /// <param name="sourceLineNumber">Line number in the source file at which the method is called.</param>
+#else   
+        /// <summary>
+        /// Writes the specified diagnostic message.
+        /// </summary>
+        /// <param name="logEvent">Log event.</param>
+#endif
+        public void Log(LogEventInfo logEvent
+#if NET4_5
+           , [CallerMemberName] string memberName = ""
+           , [CallerFilePath] string sourceFilePath = ""
+           , [CallerLineNumber] int sourceLineNumber = 0
+#endif
+            )
         {
             if (this.IsEnabled(logEvent.Level))
             {
+#if NET4_5
+                logEvent.CallerMethodName = memberName;
+                logEvent.CallerSourceFilePath = sourceFilePath;
+                logEvent.CallerLineNumber = sourceLineNumber;
+#endif
```

Comment:
Is there an alternative to capture this info only on some events? 
We should also consider not breaking compatibility

As ilya-g said:
>Consider, for example, an assembly compiled against NLog for .NET 4.0 that is being redirected to NLog for .NET 4.5 upon load — it will fail to find the method Log(LogEventInfo)